plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/AppEngineStandardFacet.java

```diff
@@ -169,19 +173,51 @@ public static void installAppEngineFacet(IFacetedProject facetedProject,
     if (facetedProject.hasProjectFacet(FACET)) {
       return;
     }
+
+    final IFacetedProjectWorkingCopy fpjwc = facetedProject.createWorkingCopy();
+    fpjwc.detect(subMonitor.newChild(20));
+    fpjwc.commitChanges(subMonitor.newChild(20));
```

Comment:
So this should be `.modify()` to avoid https://github.com/GoogleCloudPlatform/google-cloud-eclipse/issues/1155#issuecomment-310122465 as we discussed?